src/community/smart-data-loader/src/test/java/org/geoserver/smartdataloader/data/store/AbstractSmartDataLoaderPageTest.java

```diff
@@ -0,0 +1,218 @@
+import static org.geotools.jdbc.JDBCDataStoreFactory.PORT;
+import static org.geotools.jdbc.JDBCDataStoreFactory.USER;
+import static org.junit.Assume.assumeNotNull;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Properties;
+import org.apache.wicket.ajax.markup.html.form.AjaxCheckBox;
+import org.apache.wicket.markup.html.form.DropDownChoice;
+import org.apache.wicket.markup.html.form.TextField;
+import org.apache.wicket.util.tester.FormTester;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.DataStoreInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.smartdataloader.JDBCFixtureHelper;
+import org.geoserver.smartdataloader.data.SmartDataLoaderDataAccessFactory;
+import org.geoserver.smartdataloader.metadata.jdbc.utils.JdbcUrlSplitter;
+import org.geoserver.web.GeoServerWicketTestSupport;
+import org.geoserver.web.data.store.DataAccessNewPage;
+import org.geoserver.web.data.store.panel.WorkspacePanel;
+import org.geotools.data.DataAccessFactory;
+import org.geotools.data.postgis.PostgisNGDataStoreFactory;
+import org.geotools.jdbc.JDBCTestSetup;
+import org.geotools.test.FixtureUtilities;
+import org.junit.After;
+import org.junit.Test;
+import org.postgresql.jdbc.SslMode;
+
+/**
+ * Implementation of GeoServerWicketTestSupport for SmartAppSchema Postgis tests, including
+ * Geoserver and Wicket support.
+ *
+ * @author Jose Macchi - GeoSolutions
+ */
+public abstract class AbstractSmartDataLoaderPageTest extends GeoServerWicketTestSupport {
+
+    private DataAccessFactory dataStoreFactory = new SmartDataLoaderDataAccessFactory();
+
+    protected Properties fixture;
+
+    private DataStoreInfo dataStore;
+
+    private JDBCTestSetup setup;
+
+    private static final String SIMPLE_DATA_STORE_NAME = "simple-data-store";
+
+    private JDBCFixtureHelper fixtureHelper;
+
+    public AbstractSmartDataLoaderPageTest(JDBCFixtureHelper fixtureHelper) {
+        this.fixtureHelper = fixtureHelper;
+    }
+
+    protected DataAccessNewPage startPage() {
+        login();
+        final DataAccessNewPage page = new DataAccessNewPage(dataStoreFactory.getDisplayName());
+        tester.startPage(page);
+        return page;
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        this.fixture = getFixture();
+        assumeNotNull(fixture);
+        this.setup = createTestSetup();
+        setup.setFixture(fixture);
+        setup.setUp();
+        // setup geoserver datadir
+        setupGeoServerTestData();
+    }
+
+    @Override
+    protected void onTearDown(SystemTestData testData) throws Exception {
+        if (setup != null) setup.tearDown();
+    }
+
+    @After
+    public void cleanUp() {
+        if (dataStore != null) getCatalog().remove(dataStore);
```

Comment:
Move this line to onTearDown method to avoid closing the datastore on the firts test run.

Reply:
fixed